src/libs/actions/Transaction.ts

```diff
@@ -605,6 +605,10 @@ function changeTransactionsReport(transactionIDs: string[], reportID: string) {
     const transactions = transactionIDs.map((id) => allTransactions?.[id]).filter((t): t is NonNullable<typeof t> => t !== undefined);
     const transactionIDToReportActionAndThreadData: Record<string, TransactionThreadInfo> = {};
     const updatedReportTotals: Record<string, number> = {};
+    const existingSelfDMReportID = findSelfDMReportID();
+    const currentTime = DateUtils.getDBTime();
+    const selfDMReport = buildOptimisticSelfDMReport(currentTime);
+    const selfDMCreatedReportAction = buildOptimisticCreatedReportAction(currentUserEmail ?? '', currentTime);
```

Comment:
This doesn't seem right. We only want to create the selfDM if there's no existing one already